# WB001: EventBoundary on record

## Description
The `EventBoundary` attribute should be applied to classes rather than records for better extensibility and source generation compatibility.

## Why This Matters
Records in C# are immutable by design and cannot be extended with partial methods. The source generators in WabbitBot rely on partial method generation to inject event publishing logic. Using records prevents the generators from properly extending the class with the necessary event handling code.

## How to Fix
Change the record to a class:

```csharp
// ❌ Incorrect
[EventBoundary]
public record MyEvent(string Data);

// ✅ Correct
[EventBoundary]
public class MyEvent
{
    public string Data { get; init; }
}
```

## Additional Notes
- If you need immutability, consider using init-only properties on classes
- Records can still be used for simple data transfer objects that don't need event boundaries
