# WB006: Invalid cache size

## Description
The `MaxCacheSize` property in `EntityMetadata` must be a positive integer value.

## Why This Matters
Cache size limits are crucial for:
- Preventing memory exhaustion
- Ensuring predictable performance
- Proper resource management
- Avoiding negative or zero cache sizes that would disable caching

## How to Fix
Provide a positive integer for the cache size:

```csharp
// ❌ Incorrect
[EntityMetadata(TableName = "players", MaxCacheSize = 0)]
[EntityMetadata(TableName = "players", MaxCacheSize = -1)]
[EntityMetadata(TableName = "players", MaxCacheSize = null)]

// ✅ Correct
[EntityMetadata(TableName = "players", MaxCacheSize = 100)]
[EntityMetadata(TableName = "players", MaxCacheSize = 1000)]
```

## Additional Notes
- Cache sizes should be based on expected working set size
- Consider memory usage and access patterns when setting limits
- Start with reasonable defaults and adjust based on monitoring
