# WB005: Missing TableName in EntityMetadata

## Description
Entity classes decorated with `[EntityMetadata]` must specify a `TableName` property for proper database table mapping.

## Why This Matters
The `TableName` is required for:
- Entity Framework Core to map classes to database tables
- Source generators to create proper repository and service code
- Maintaining consistency between code and database schema
- Preventing runtime mapping errors

## How to Fix
Add the `TableName` property to your `EntityMetadata` attribute:

```csharp
// ❌ Incorrect
[EntityMetadata]
public class Player
{
    public Guid Id { get; set; }
    public string Name { get; set; }
}

// ✅ Correct
[EntityMetadata(TableName = "players")]
public class Player
{
    public Guid Id { get; set; }
    public string Name { get; set; }
}
```

## Additional Notes
- Table names should be lowercase and plural
- Use underscores for multi-word table names (snake_case)
- Ensure table names match your database schema
