# WB004: Event class without IEvent interface

## Description
All event classes must implement the `IEvent` interface to ensure proper event bus routing and metadata handling.

## Why This Matters
The `IEvent` interface provides essential metadata (EventId, Timestamp, EventBusType) that the event bus needs for:
- Event correlation and tracking
- Proper routing between event buses
- Debugging and monitoring
- Ensuring event immutability

## How to Fix
Implement the `IEvent` interface on your event classes:

```csharp
// ❌ Incorrect
public class MatchStartedEvent
{
    public Guid MatchId { get; init; }
}

// ✅ Correct
public class MatchStartedEvent : IEvent
{
    public Guid EventId { get; init; } = Guid.NewGuid();
    public DateTime Timestamp { get; init; } = DateTime.UtcNow;
    public EventBusType EventBusType { get; init; } = EventBusType.Core;

    public Guid MatchId { get; init; }
}
```

## Additional Notes
- Event classes should be immutable records when possible
- The `EventId` should be unique for each event instance
- `Timestamp` should use UTC time
- `EventBusType` determines which event bus handles the event (Core, Discord, or Global)
