# WB003: CRUD event detected

## Description
Events should not be used for Create, Read, Update, or Delete (CRUD) operations. Database operations must be handled through repositories, not the event bus.

## Why This Matters
The event system is designed for communication and decoupling, not for direct database operations. Mixing CRUD operations with events violates the separation of concerns and can lead to:
- Race conditions
- Inconsistent data states
- Difficult debugging
- Performance issues

## How to Fix
Use repositories for database operations instead of events:

```csharp
// ❌ Incorrect - Don't publish CRUD events
[EventBoundary]
public class UserService
{
    public async Task CreateUserAsync(User user)
    {
        var createdEvent = new UserCreatedEvent(user.Id);
        await _eventBus.PublishAsync(createdEvent); // Wrong approach
    }
}

// ✅ Correct - Use repositories directly
[EventBoundary]
public class UserService
{
    private readonly IUserRepository _userRepository;

    public UserService(IUserRepository userRepository, ICoreEventBus eventBus)
    {
        _userRepository = userRepository;
    }

    public async Task CreateUserAsync(User user)
    {
        await _userRepository.AddAsync(user); // Direct repository call

        // Optionally publish domain events after successful persistence
        var createdEvent = new UserCreatedEvent(user.Id);
        await _eventBus.PublishAsync(createdEvent);
    }
}
```

## Additional Notes
- Events should represent business facts or commands, not database operations
- Use repositories for all data persistence
- Events can be published *after* successful repository operations to notify other parts of the system
